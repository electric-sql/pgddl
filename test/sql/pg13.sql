\i test/sql/pg12.sql

alter view test_class_v rename column a to aardvark;
select ddlx_create('test_class_v'::regclass);

