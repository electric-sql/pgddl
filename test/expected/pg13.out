\pset null _null_
SET client_min_messages = warning;
SET ROLE postgres;
create table tab_generated (
       a integer generated by default as identity,
       b integer generated always as identity,
       c integer generated always as (a+b) stored,
       d integer default 42,
       e serial,
       f integer
);
-- select * from ddlx_describe('tab_generated'::regclass) d;
\pset format unaligned
select ddlx_script('tab_generated');
ddlx_script
BEGIN;

/*
ALTER TABLE tab_generated ALTER e DROP DEFAULT;

DROP TABLE tab_generated;
*/

-- Type: TABLE ; Name: tab_generated; Owner: postgres

CREATE TABLE tab_generated (
    a integer GENERATED BY DEFAULT AS IDENTITY,
    b integer GENERATED ALWAYS AS IDENTITY,
    c integer GENERATED ALWAYS AS (a + b) STORED,
    d integer,
    e integer NOT NULL,
    f integer
);

ALTER TABLE tab_generated OWNER TO postgres;

ALTER TABLE tab_generated ALTER d SET DEFAULT 42;

ALTER SEQUENCE tab_generated_e_seq OWNED BY tab_generated.e;

COMMENT ON TABLE tab_generated IS NULL;

-- DEPENDANTS

ALTER TABLE tab_generated ALTER e SET DEFAULT nextval('tab_generated_e_seq'::regclass);

END;

(1 row)
